pandas/tests/io/pytables/test_store.py

```diff
@@ -79,33 +77,25 @@ def test_format_kwarg_in_constructor(self, setup_path):
 
         msg = "format is not a defined argument for HDFStore"
 
-        with ensure_clean_path(setup_path) as path:
+        with tm.ensure_clean(setup_path) as path:
             with pytest.raises(ValueError, match=msg):
                 HDFStore(path, format="table")
 
     def test_context(self, setup_path):
-        path = create_tempfile(setup_path)
-        try:
-            with HDFStore(path) as tbl:
-                raise ValueError("blah")
-        except ValueError:
-            pass
-        finally:
-            safe_remove(path)
-
-        try:
+        with tm.ensure_clean(setup_path) as path:
+            try:
+                with HDFStore(path) as tbl:
+                    raise ValueError("blah")
+            except ValueError:
+                pass
```

Comment:
can you wrap L91 and L93 with another ensure_clean (separate ones)

Reply:
Done 👍

Reply:
I have wrapped the remaining lines in an additional ensure_clean. I think the following lines all needed to come under the same ensure_clean rather than separate ones otherwise the assertions fail:

```
tbl["a"] = tm.makeDataFrame()
assert len(tbl) == 1
assert type(tbl["a"]) == DataFrame
```